File: cmd/gui/types/crop.go
package types

// CropInfo is the per-file flip/rotate/crop the user applied in the Crop/Rotate modal. It is applied to the source image
// (flip → rotate → crop) before any enhancement runs. A zero value (Width <= 0 || Height <= 0) means "no crop".
type CropInfo struct {
	Rotation float64
	FlipH    bool
	FlipV    bool
	Top      int
	Left     int
	Width    int
	Height   int
}

File: cmd/gui/frontend/src/utils/crop.ts
import type { CropInfo } from '@/bindings/gui/types';

/**
 * A stable cache-key fragment for a crop; empty string when there's no crop so uncropped keys stay unchanged.
 *
 * Shared by the image/enhanced-image caches and the face-detection cache so all three key consistently on the crop.
 *
 * @param crop - The crop to serialize, or undefined for no crop.
 */
export const cropToken = (crop?: CropInfo) =>
    crop
        ? `_c${crop.Rotation}-${crop.FlipH ? 1 : 0}${crop.FlipV ? 1 : 0}-${crop.Left}-${crop.Top}-${crop.Width}-${crop.Height}`
        : '';
